ocs_ci/ocs/warp.py

```diff
@@ -0,0 +1,186 @@
+        pvc_size = "50Gi"
+        self.pod_name = "warp-pod"
+        self.pvc_obj = helpers.create_pvc(
+            sc_name=constants.DEFAULT_STORAGECLASS_RBD,
+            namespace=self.namespace,
+            size=pvc_size,
+        )
+        self.pod_obj = helpers.create_pod(
+            constants.CEPHBLOCKPOOL,
+            namespace=self.namespace,
+            pod_name=self.pod_name,
+            pvc_name=self.pvc_obj.name,
+            sa_name=self.sa_name,
+            pod_dict_path=self.pod_dic_path,
+            dc_deployment=True,
+            deploy_pod_status=constants.STATUS_COMPLETED,
+        )
+
+    def install_warp(self, timeout=1800):
+        """
+        Install minio warp S3 benchmark:
+        https://github.com/minio/warp
+
+        """
+        # Install warp
+        log.info(f"Installing warp S3 benchmark on testing pod {self.pod_obj.name}")
+        warp_path = "https://github.com/minio/warp/releases/download/v0.5.5/warp_0.5.5_Linux_x86_64.tar.gz"
+        self.pod_obj.exec_cmd_on_pod(f"wget {warp_path}", timeout=timeout)
+        self.pod_obj.exec_cmd_on_pod(
+            "tar -xzvf warp_0.5.5_Linux_x86_64.tar.gz", timeout=timeout
+        )
+        log.info("Successfully installing warp s3 benchmark")
+
+    def run_benchmark(
+        self,
+        bucket_name=None,
+        access_key=None,
+        secret_key=None,
+        duration=None,
+        concurrent=None,
+        objects=None,
+        obj_size=None,
+        timeout=None,
+        validate=True,
+    ):
+        """
+         Running Warp S3 benchmark
+         Usage detail can be found at: https://github.com/minio/warp
+
+        Args:
+            bucket_name (string): Name of bucket
+            access_key (string): Access Key credential
+            secret_key (string): Secret Key credential
+            duration (string): Duration of the test
+            concurrent (int): number of concurrent
+            objects (int): number of objects
+            obj_size (int): size of object
+            timeout (int): timeout in seconds
+            validate (Boolean): Validates whether running workload is completed.
+
+        """
+
+        # Running warp S3 benchmark
+        log.info("Running Minio Warp S3 benchmark")
+        timeout = timeout if timeout else 3600
+        self.access_key = access_key if access_key else self.access_key
+        self.secret_key = secret_key if secret_key else self.secret_key
+        self.bucket_name = bucket_name if bucket_name else self.bucket_name
+        self.duration = duration if duration else self.duration["duration"]
+        self.concurrent = concurrent if concurrent else self.concurrent["concurrent"]
+        self.objects = objects if objects else self.objects["objects"]
+        self.obj_size = obj_size if obj_size else self.obj_size["obj.size"]
+        self.pod_obj.exec_cmd_on_pod(
+            f"{self.warp_bin_dir} mixed "
```

Comment:
`/home/warp/warp` would be considered as directory I suppose? Hence this might fail.

Reply:
Yes, 'self.warp_bin_dir' is /home/warp/warp which is pre-defined in my image quay.io/tiffanyn108/warp:latest which is the path where the test running take place.

Reply:
Understood. what you're doing here is run warp on pod like this... `/home/warp/warp mixed ...` but I feel `/home/warp/warp` is considered as directory not executable hence it will fail. To confirm, you can test the same locally on your machine if you have warp installed. Alternatively, you can also set `/home/warp/` as current working directory in your warp image and here in the test you can directly run it as warp. 
If you think this will work, its better to run validation and check :D

Reply:
@mashetty330 '/home/warp/ is a directory and 'warp' is a binary.  I've tested locally already.  It works as expected, I will start a Jenkins validation soon.

Reply:
@tiffanyn108 sounds good to me if it works as binary :+1: